pages/search/advanced.tsx

```diff
@@ -0,0 +1,225 @@
+      </Head>
+      {alert && (
+        <Notification
+          notificationType="warning"
+          notificationContent={
+            <>Please enter at least one field to submit an advanced search.</>
+          }
+        />
+      )}
+      <Heading level="two">Advanced Search</Heading>
+      <Form
+        id="advancedSearchForm"
+        method="post"
+        action={`${BASE_URL}/search`}
+        onSubmit={handleSubmit}
+      >
+        <FormRow gap="grid.m">
+          <FormField id="advancedSearchLeft" gap="grid.s">
+            {textInputFields.map(({ name, key, label }) => {
+              return (
+                <TextInput
+                  id={name}
+                  labelText={label}
+                  type="text"
+                  name={name}
+                  value={searchFormState[name]}
+                  key={key}
+                  onChange={(e) => handleInputChange(e, "input_change")}
+                />
+              )
+            })}
+            <Select
+              id="languageSelect"
+              name="language"
+              labelText="Language"
+              aria-labelledby="languageSelect-label"
+              value={searchFormState["selectedFilters"].language}
+              onChange={(e) => handleInputChange(e, "filter_change")}
+            >
+              {languageOptions.map((language) => {
+                return (
+                  <option value={language.value} key={language.value}>
+                    {language.label}
+                  </option>
+                )
+              })}
+            </Select>
+          </FormField>
+          <FormField id="advancedSearchRight" gap="grid.s">
+            <Fieldset legendText="Date" id="dates">
+              <FormRow id="dates" gap="grid.m">
+                <FormField>
+                  <DatePicker
+                    id="dateAfter"
+                    nameFrom="dateAfter"
+                    dateType="year"
+                    labelText="From"
+                    helperText="e.g. 1901"
+                    initialDate={searchFormState["selectedFilters"].dateAfter}
+                    onChange={(e) => handleDateChange("dateAfter", e)}
+                  />
+                </FormField>
+                <FormField>
+                  <DatePicker
+                    id="dateBefore"
+                    nameFrom="dateBefore"
+                    dateType="year"
+                    labelText="To"
+                    helperText="e.g. 2000"
+                    initialDate={searchFormState["selectedFilters"].dateBefore}
+                    onChange={(e) => handleDateChange("dateBefore", e)}
+                  />
+                </FormField>
+              </FormRow>
+            </Fieldset>
+            <CheckboxGroup
+              id="formats"
+              name="formats"
+              labelText="Formats"
+              onChange={handleCheckboxChange}
+              value={searchFormState["selectedFilters"].materialType}
```

Comment:
This should be `defaultValue` to be controlled. Then you can remove the `name` from the individual Checkbox component on line 197. For more info: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-form-elements-checkboxgroup--docs

Reply:
I believe value has to be set in order to clear the values based on the form state.

Reply:
I was able to remove the name on the individual checkboxes now that CheckboxGroup was added, however.